ray-operator/api/raycluster/v1alpha1/raycluster_types.go

```diff
@@ -62,6 +65,18 @@ type ScaleStrategy struct {
 	WorkersToDelete []string `json:"workersToDelete,omitempty"`
 }
 
+// JobSpec is the spec for the job which will be submitted when raycluster is ready
+type JobSpec struct {
+	// Dict to setup execution environment, it is variable, the map value is expected a serialized value.
+	RuntimeEnv map[string]string `json:"runtimeEnv,omitempty"`
+	// Command to start execution, ex: "python script.py"
+	Entrypoint string `json:"entrypoint,omitempty"`
+	// Metadata to pass in to configure job behavior or use as tags
+	Metadata map[string]string `json:"metadata,omitempty"`
+	// ShutdownAfterJobFinished indicates whether delete raycluster after job finish
+	ShutdownAfterJobFinished bool `json:"shutdownAfterJobFinished,omitempty"`
```

Comment:
Great. This would be helpful for debug purpose.  

I am also thinking whether we need some time for other components like log collector etc to persist artifacts, etc.